client/app/services/retire-service-modal/retire-service-modal.component.js

```diff
@@ -60,7 +61,13 @@ function ComponentController ($scope, $state, CollectionsApi, EventNotifications
       })
     })
   }
-
+  function isDisabled () {
+    if (lodash.isObject(vm.modalData.date)) {
```

Comment:
`return lodash.isObject(vm.modalData.date)`?

(unless you really have an `if (bool) { return true; } else { return false; }` policy :))